### PR TITLE
cut_block_by_epochs give a new Block now.

### PR DESCRIPTION
**Pull request by samuelgarcia**
Now cut_block_by_epochs give a new block and don't do a inplace replacement of segments.
Similarly to cut_segment_by_epoch.

See #755 

@mdenker @JuliaSprenger : could you check this ?
I think you are originals authors no ?

### CODE REVIEW

File: neo/utils.py [hunk 1]
```diff
@@ -490,10 +491,11 @@ def cut_block_by_epochs(block, properties=None, reset_time=False):
         for epoch in epochs:
             new_segments = cut_segment_by_epoch(
                 seg, epoch=epoch, reset_time=reset_time)
-            block.segments += new_segments
```
Comment (JuliaSprenger) [inline neo/utils.py:493]: is there a reason to use `+=` rather than `extend` here?

Comment (samuelgarcia) [reply neo/utils.py:493]: I am anticipating the day wher we will have pseudo list for relationship.
I guess append and extend will be part of methods not sure for +=

Comment (JuliaSprenger) [reply neo/utils.py:493]: then it would be better to stick to `extend`, no?

**Comment by samuelgarcia** [comment]
Ready to merge.
Should close #755

**Pull request merged**